app/src/test/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModelTest.kt

```diff
@@ -26,233 +26,238 @@ import com.wire.android.config.TestDispatcherProvider
-        editGuestAccessViewModel.editGuestAccessState =
-            editGuestAccessViewModel.editGuestAccessState.copy(isGuestAccessAllowed = true)
-
-        editGuestAccessViewModel.updateGuestAccess(false)
-
-        assertEquals(true, editGuestAccessViewModel.editGuestAccessState.shouldShowGuestAccessChangeConfirmationDialog)
-        coVerify(inverse = true) {
-            updateConversationAccessRoleUseCase(any(), any(), any())
-        }
-    }
-
-    @Test
-    fun `given useCase runs with success, when_generating guest link, then invoke it once`() = runTest {
-        coEvery {
-            generateGuestRoomLink.invoke(any(), any())
-        } returns GenerateGuestRoomLinkResult.Success
-
-        editGuestAccessViewModel.onRequestGuestRoomLink()
-        coVerify(exactly = 1) {
-            generateGuestRoomLink.invoke(any(), null)
-        }
-        assertEquals(false, editGuestAccessViewModel.editGuestAccessState.isGeneratingGuestRoomLink)
-    }
-
-    @Test
-    fun `given useCase runs with failure, when generating guest link, then show dialog error`() = runTest {
-        coEvery {
-            generateGuestRoomLink(any(), any())
-        } returns GenerateGuestRoomLinkResult.Failure(NetworkFailure.NoNetworkConnection(null))
-
-        editGuestAccessViewModel.onRequestGuestRoomLink()
-
-        coVerify(exactly = 1) {
-            generateGuestRoomLink(any(), null)
+            assertEquals(true, editGuestAccessViewModel.editGuestAccessState.isGuestAccessAllowed)
         }
-        assertEquals(true, editGuestAccessViewModel.editGuestAccessState.isFailedToGenerateGuestRoomLink)
-    }
-
-    @Test
-    fun `given useCase runs with success, when revoking guest link, then invoke it once`() = runTest {
-        coEvery {
-            revokeGuestRoomLink(any())
-        } returns RevokeGuestRoomLinkResult.Success
 
-        editGuestAccessViewModel.removeGuestLink()
-
-        coVerify(exactly = 1) {
-            revokeGuestRoomLink(any())
+//    @Test
+//    fun `given a failure when running updateConversationAccessRole, when trying to enable guest access, then do not enable guest access`() {
+//        editGuestAccessViewModel.editGuestAccessState =
+//            editGuestAccessViewModel.editGuestAccessState.copy(isGuestAccessAllowed = false)
+//        coEvery {
+//            updateConversationAccessRoleUseCase(any(), any(), any())
+//        } returns UpdateConversationAccessRoleUseCase.Result.Failure(
+//            CoreFailure.MissingClientRegistration
+//        )
+//
+//        editGuestAccessViewModel.updateGuestAccess(true)
+//
+//        coVerify(exactly = 1) {
+//            updateConversationAccessRoleUseCase(any(), any(), any())
+//        }
+//        assertEquals(
+//            false,
+//            editGuestAccessViewModel.editGuestAccessState.isGuestAccessAllowed
+//        )
+//    }
```

Comment:
Should it still be commented out?

Reply:
You are right, these tests were broken, because we were not running them 😅  Since junit4 annotations in tests were skipped. So I'm fixing them

Reply:
Solved :D